s2fft/precompute_transforms/custom_ops.py

```diff
@@ -0,0 +1,410 @@
+from functools import partial
+from typing import Tuple
+
+import jax.numpy as jnp
+import numpy as np
+from jax import jit
+
+
+def wigner_subset_to_s2(
+    flmn: np.ndarray,
+    spins: np.ndarray,
+    DW: Tuple[np.ndarray, np.ndarray],
+    L: int,
+    sampling: str = "mw",
+) -> np.ndarray:
+    r"""
+    Transforms an arbitrary subset of Wigner coefficients onto a subset of spin signals
+    on the sphere.
+
+        This function takes a collection of spin spherical harmonic coefficients each with
+        a different (though not necessarily unique) spin and maps them back to their
+        corresponding pixel-space representations. Following this operation one may
+        liftn this collection of spin signals to a signal on SO(3) by exploiting the
+        correct Mackey functions.
+
+    Args:
+        flmn (np.ndarray): Collection of spin spherical harmonic coefficients
+            with shape :math:`[batch, n_s, L, 2L-1, channels]`.
+        spins (np.ndarray): Spins of each field in fs with shape :math:`[n_s]`.
+        DW (Tuple[np.ndarray, np.ndarray]): Fourier coefficients of the reduced
+            Wigner d-functions and the corresponding upsampled quadrature weights.
+        L (int): Harmonic band-limit.
+        sampling (str, optional): Sampling scheme.  Supported sampling schemes include
+            {"mw", "mwss"}. Defaults to "mw".
+
+    Returns:
+        np.ndarray: A collection of spin signals with shape :math:`[batch, n_s, n_\theta, n_\phi, channels]`.
+
+    """
+    if sampling.lower() not in ["mw", "mwss"]:
+        raise ValueError(
+            f"Fourier-Wigner algorithm does not support {sampling} sampling."
+        )
+
```

Comment:
We should check the input shapes are as expected, e.g. that flmn and spins have the same n_s and that shapes match the L provided.